### ISSUE TITLE
[packagers] Flatpak ignores repository name property

### ISSUE THREAD

**Issue by crschnick** [jreleaser/jreleaser#1323]
My flatpak publish always creates a git repository with the name of `componentId`, even though I set a different name under `flatpak.repository.name`. Is it required that the repository name is equal to the componentId?

**Comment by aalmiray** [comment]
No. It's a requirement for publishing to flathub but not to a custom flatpak repository.

**Comment by aalmiray** [comment]
Cannot reproduce at the moment. Given this configuration

```groovy
        app {
            flatpak {
                active = 'ALWAYS'
                continueOnError = true
                componentId = 'com.acme.app'
                developerName = 'Acme'
                runtime = 'FREEDESKTOP'
                runtimeVersion = '21.08'
                category('Developer Tools')
                repository {
                    active = 'RELEASE'
                    name = 'foobar'
                    branch = 'master'
                }
            }
        }
```

The resolve model shows

```
        flatpak:
            enabled: true
            active: ALWAYS
            continueOnError: true
            templateDirectory: src/jreleaser/distributions/app/flatpak
            commitAuthor:
                name: jreleaserbot
                email: jreleaser@kordamp.org
            componentId: com.acme.app
            categories:
                Developer Tools
            developerName: Acme
            screenshots:
                screenshot 0:
                    type: SOURCE
                    url: https://jreleaser.org/images/app-full-release-command.png
                    primary: true
                    caption: JReleaser in action
            icons:
                icon 0:
                    url: https://raw.githubusercontent.com/jreleaser/jreleaser/main/src/media/icon_256x256.png
                    width: 256
                    height: 256
                    primary: true
            runtime: FREEDESKTOP
            runtimeVersion: 21.08
            sdkExtensions:
                org.freedesktop.Sdk.Extension.openjdk
            finishArgs:
                --env=PATH=/app/jre/bin:/usr/bin:/app/bin
                --env=JAVA_HOME=/app/jre
            repository:
                enabled: true
                active: RELEASE
                owner: jreleaser
                name: foobar
                tagName: v{{projectVersion}}
                branch: master
                branchPush: HEAD
                username: jreleaser
                token: ************
                commitMessage: {{distributionName}} {{tagName}}
```

The name of the repository is set to `foobar` instead of `com.acme.app`.

**Comment by crschnick** [comment]
I have this configuration currently:

    jreleaser {
        packagers {
            flatpak {
                active = 'ALWAYS'
                continueOnError = false
                templateDirectory = 'flatpak'
                commitAuthor {
                    name = 'crschnick'
                    email = 'crschnick@xpipe.io'
                }
    
                repository {
                    active = 'ALWAYS'
                    repoOwner = 'xpipe-io'
                    name = 'flatpak'
                    username = 'crschnick'
                    token = proj.hasProperty("XPIPE_GITHUB_TOKEN") ? proj.property("XPIPE_GITHUB_TOKEN") : System.getenv("XPIPE_GITHUB_TOKEN")
                    commitMessage = '{{version}}'
                    tagName = '{{tagName}}'
                }
    
                componentId = 'io.xpipe.app'
                category('Development')
                developerName = 'Christopher Schnick'
                runtime = 'FREEDESKTOP'
                runtimeVersion = '21.08'
    
                finishArg('--share=network')
            }
        }
    }

When I run the publish task, this is the output:

    [INFO]  Publishing distributions
    [INFO]    - Publishing api distribution
    [INFO]    - Publishing beacon distribution
    [INFO]    - Publishing core distribution
    [INFO]    - Publishing portable distribution
    [INFO]      [flatpak] publishing portable distribution
    [INFO]      [flatpak] setting up repository xpipe-io/io.xpipe.app
    [INFO]      [flatpak] pushing to xpipe-io/io.xpipe.app
    [INFO]  Writing output properties to build/jreleaser/output.properties
    [INFO]  JReleaser succeeded after 5.184 s

**Comment by aalmiray** [comment]
What's the output of the resolved model? You can see it inside `trace.log` or by invoking `gradle jreleaserConfig`.

**Comment by crschnick** [comment]
The model resolves it correctly:

        flatpak:
            enabled: true
            active: ALWAYS
            continueOnError: false
            templateDirectory: /home/crschnick/IdeaProjects/xpipex/dist/flatpak
            commitAuthor:
                name: crschnick
                email: crschnick@xpipe.io
            componentId: io.xpipe.app
            categories:
                Development
            developerName: Christopher Schnick
            screenshots:
                screenshot 0:
                    type: SOURCE
                    url: https://user-images.githubusercontent.com/72509152/230100929-4476f76c-ea81-43d9-ac4a-b3b02df2334e.png
                    primary: true
                    caption: App in action
                    width: 932
                    height: 531
            icons:
                icon 0:
                    url: https://github.com/xpipe-io/xpipe/raw/master/dist/logo/logo.png
                    width: 128
                    height: 128
                    primary: true
            runtime: FREEDESKTOP
            runtimeVersion: 21.08
            finishArgs:
                --share=network
            repository:
                enabled: true
                active: ALWAYS
                owner: xpipe-io
                name: flatpak
                tagName: {{tagName}}
                branch: HEAD
                branchPush: HEAD
                username: crschnick
                token: ************
                commitMessage: {{version}}

**Comment by aalmiray** [comment]
I get this from the `trace.log` file reported at #1324 

```
        flatpak:
            enabled: true
            active: ALWAYS
            continueOnError: false
            templateDirectory: /home/crschnick/IdeaProjects/xpipex/dist/flatpak
            commitAuthor:
                name: crschnick
                email: crschnick@xpipe.io
            componentId: io.xpipe.app
            categories:
                Development
            developerName: Christopher Schnick
            screenshots:
                screenshot 0:
                    type: SOURCE
                    url: https://user-images.githubusercontent.com/72509152/230100929-4476f76c-ea81-43d9-ac4a-b3b02df2334e.png
                    primary: true
                    caption: App in action
                    width: 932
                    height: 531
            icons:
                icon 0:
                    url: https://github.com/xpipe-io/xpipe/raw/master/dist/logo/logo.png
                    width: 128
                    height: 128
                    primary: true
            runtime: FREEDESKTOP
            runtimeVersion: 21.08
            finishArgs:
                --share=network
            repository:
                enabled: true
                active: ALWAYS
                owner: xpipe-io
                name: io.xpipe.app
                tagName: {{tagName}}
                branch: HEAD
                branchPush: HEAD
                username: crschnick
                token: ************
                commitMessage: {{version}}
```

Running my own test with the configuration shown earlier in this thread I get

```
> Task :jreleaserPublish
[INFO]  JReleaser 1.5.1
[INFO]    - basedir set to /Users/aalmiray/dev/github/smoketests-jreleaser
[INFO]    - outputdir set to /Users/aalmiray/dev/github/smoketests-jreleaser/build/jreleaser
[INFO]  git-root-search set to false
[WARN]  Auto configure detected 'smoketests-jreleaser' as repository name but project has 'jreleaser' configured
[INFO]  Loading variables from /Users/aalmiray/.jreleaser/config.toml
[INFO]  Loading variables from /Users/aalmiray/dev/github/smoketests-jreleaser/.env
[INFO]  Validating configuration
[INFO]  Strict mode set to false
[INFO]  Project version set to 1.0.0
[INFO]  Release is not snapshot
[INFO]  Timestamp is 2023-04-15T09:07:54.40993+02:00
[INFO]  HEAD is at 0d4f8f4
[INFO]  Platform is osx-x86_64
[INFO]  dry-run set to true
[INFO]  Executing before hook(s): 1 in total
[INFO]  Calculating checksums for distributions and files
[INFO]    [checksum] All checksums are up-to-date. Skipping
[INFO]  Executing before hook(s): 1 in total
[INFO]  Preparing distributions
[INFO]    - Preparing app distribution
[INFO]      [flatpak] preparing app distribution
[INFO]  Executing before hook(s): 1 in total
[INFO]  Packaging distributions
[INFO]    - Packaging app distribution
[INFO]      [flatpak] packaging app distribution
[INFO]  Executing before hook(s): 1 in total
[INFO]  Publishing distributions
[INFO]    - Publishing app distribution
[INFO]      [flatpak] publishing app distribution
[INFO]      [flatpak] setting up repository jreleaser/foobar
[INFO]  Writing output properties to build/jreleaser/output.properties
[INFO]  JReleaser succeeded after 2.044 s
```

I must be missing something.

**Comment by crschnick** [comment]
I think the hierarchy differs. The wrong repository name is shown in the trace at distributions.portable.flatpak. The correct name (the excerpt that I posted in this thread earlier) is listed under packagers.flatpak

**Comment by aalmiray** [comment]
Ahh gotcha. I think I know where the problem lies.

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.6.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.6.0))
